### PR TITLE
Improve hint and error ARIA logic in Form controls

### PR DESCRIPTION
**Pull request by robpataki**
- Update **TextInput**, **TextArea** and **FormGroup** (which automatically fixes **DateInput**, **Radios** and **Checkboxes**) components
- In above components the internal `aria-describedby` attribute only references the **hint** element by ID when it actually exists. This allows hint to be truly optional eliminating HTML validation and accessibility issues currently happening.
- Add logic in the same components to reference the error message by ID for `aria-describedby` when error message exists. When error message and hint both are present, referencing error takes priority.

I haven't added any new tests, but manually testing the components in Storybook, I can verify that  both issues raised in https://github.com/daniel-ac-martin/NotGovUK/issues/656 are fixed and the components work as expected.

### CODE REVIEW

File: components/form-group/src/FormGroup.tsx [hunk 1]
```diff
@@ -35,6 +35,9 @@ export const FormGroup: FC<FormGroupProps> = ({
   ];
   const classes = classBuilder('govuk-form-group', classBlock, classModifiers, className);
   const hintId = _hintId || `${id}-hint`;
+  const errorId = `${id}-error`;
+  const describedById = error ? errorId : hint && hintId;
```
Comment (daniel-ac-martin) [inline components/form-group/src/FormGroup.tsx:39]: Looking at..: https://design-system.service.gov.uk/components/text-input/#text-input-with-errors-example-html
I don't think this fixes the bug completely. - It looks like this logic will _either_ reference the hint _or_ reference the error message. Whereas it seems that, if both are present, we should reference both.

On that note, I think we should name this constant as simply `describedBy`, as it is not necessarily a single ID.
```suggestion
  const describedBy = ([
    hint && hintId,
    error && errorId
  ]
    .filter(e => e)
    .join(' ') || undefined
  );
```
Perhaps? (There's probably a nicer way to lay that out!)

**Review by daniel-ac-martin — CHANGES_REQUESTED** [review]
Thanks for raising this @robpataki.
It's on the right track, but I can spot a few issues with it.

File: components/form-group/src/FormGroup.tsx (hunk 1 above)
Comment (robpataki) [reply components/form-group/src/FormGroup.tsx:39]: @daniel-ac-martin Looking at the GOVUK behaviour, it does seem like they are using both when set. Your suggestion makes sense.

File: components/form-group/src/FormGroup.tsx [hunk 2]
```diff
@@ -35,6 +35,9 @@ export const FormGroup: FC<FormGroupProps> = ({
   ];
   const classes = classBuilder('govuk-form-group', classBlock, classModifiers, className);
   const hintId = _hintId || `${id}-hint`;
+  const errorId = `${id}-error`;
```
[the remark this answers is not in the document]
Comment (robpataki) [reply components/form-group/src/FormGroup.tsx:38]: Makes sense, I'll update the code accordingly.

**Comment by robpataki** [comment]
@daniel-ac-martin I've updated the code based on your feedback.

**Comment by robpataki** [comment]
Hello @daniel-ac-martin please let me know if you have any further comments on this PR?

**Review by daniel-ac-martin — CHANGES_REQUESTED** [review]
Looks much better. :-)

Could you just update line 52 in FormGroup.tsx to make use of the new `errorId`? i.e:
```tsx
{ !error ? null : <ErrorMessage id={`${id}-error`}>{error}</ErrorMessage>}
```
could be:
```tsx
{ !error ? null : <ErrorMessage id={errorId}>{error}</ErrorMessage>}
```

Looks good to me, otherwise.

**Comment by robpataki** [comment]
@daniel-ac-martin `errorId` is in action now :)

**Review by daniel-ac-martin — APPROVED** [review]

**Comment by robpataki** [comment]
Thanks @daniel-ac-martin for approving my PR 🙏 . I've never had to sign commits, so please bear with me whilst I try to get this PR signed so it can be merged in.

**Comment by robpataki** [comment]
@daniel-ac-martin I believe I've signed all 3 commits in this PR, over to you :)  
Screenshot 2023-01-23 at 17 24 12

**Comment by daniel-ac-martin** [comment]
Thanks @robpataki.
I've always used GPG for commit signing, but it looks like SSH works just as well. It just gives us some assurance of the code provenance and it seems like it's not too inconvenient.

I've brought this branch up to date with the master branch as I've needed to put in a couple of fixes to get some of the CI checks to pass.

Before I merge this, can I just confirm that you are happy to assign copyright on this (to the Crown)?

**Comment by robValtech** [comment]
Yay, excellent!

> Before I merge this, can I just confirm that you are happy to assign copyright on this (to the Crown)?

I would be honoured :)

**Comment by robpataki** [comment]
Yay, excellent!

> Before I merge this, can I just confirm that you are happy to assign copyright on this (to the Crown)?

I would be honoured :)

**Review by daniel-ac-martin — APPROVED** [review]

**Comment by daniel-ac-martin** [comment]
Thanks again @robpataki! :-)

**Comment by robValtech** [comment]
> Thanks again @robpataki! :-)

My pleasure, I've got a few more where this one came from :) Is it possible to know when the next release/tag that including this issue fix will be released? My team is currently working on a project using this library and I'm trying to navigate local NOTGOVUK updates and local fixes on the propject balanced.

**Comment by robpataki** [comment]
> Thanks again @robpataki! :-)

My pleasure, I've got a few more where this one came from :) Is it possible to know when the next release/tag including this issue fix will be released? My team is currently working on a project using this library and I'm trying to navigate NOTGOVUK updates and local fixes on the project balanced.

**Comment by robpataki** [comment]
Hello @daniel-ac-martin. Is there any indication when a new tag will be released including this update?

**Pull request merged**